src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -445,25 +571,39 @@ class WordHighlighter {
 
 	private renderDecorations(): void {
 		this.renderDecorationsTimer = -1;
-		const decorations: IModelDeltaDecoration[] = [];
-		for (const info of this.workerRequestValue) {
-			if (info.range) {
-				decorations.push({
-					range: info.range,
-					options: getHighlightDecorationOptions(info.kind)
-				});
+		// create new loop, iterate over current editors using this.codeEditorService.listCodeEditors(),
+		// if the URI of that codeEditor is in the map, then add the decorations to the decorations array
+		// then set the decorations for the editor
+		const currentEditors = this.codeEditorService.listCodeEditors();
+		for (const editor of currentEditors) {
+			const editorHighlighterContrib = WordHighlighterContribution.get(editor);
+			if (!editorHighlighterContrib) {
+				continue;
 			}
-		}
 
-		this.decorations.set(decorations);
-		this._hasWordHighlights.set(this.hasDecorations());
+			const newDecorations: IModelDeltaDecoration[] = [];
+			const uri = editor.getModel()?.uri;
+			if (uri && this.workerRequestValue.has(uri)) {
+				const oldDecorations: string[] | undefined = this.storedDecorations.get(uri);
```

Comment:
if `uri` exists but there is no results, we should probably clear the decorations for the `uri`